lib/ansible/modules/cloud/vmware/vmware_datastore_host.py

```diff
@@ -0,0 +1,328 @@
+        except vim.fault.AlreadyExists:
+            self.module.fail_json(msg=error_message_mount + ": The local path already exists on the host, or the remote path is already mounted on the host")
+        except vim.fault.HostConfigFault:
+            self.module.fail_json(msg=error_message_mount + ": Unable to mount the NAS volume")
+        except vmodl.fault.InvalidArgument:
+            self.module.fail_json(msg=error_message_mount + ": The datastore name is invalid, or the spec is invalid")
+        except vim.fault.NoVirtualNic:
+            self.module.fail_json(msg=error_message_mount + ": VMkernel TCPIP stack is not configured for NFS data")
+        except vim.fault.NoGateway:
+            self.module.fail_json(msg=error_message_mount + ": VMkernel gateway is not configured")
+        except Exception as e:
+            self.module.fail_json(msg=error_message_mount + ": " + str(e))
+        self.module.exit_json(changed=True, result="datastore %s on host %s" % (self.datastore_name, self.esxi_hostname))
+
+    def mount_vmfs_datastore_host(self):
+        ds_path = "/vmfs/devices/disks/" + str(self.vmfs_device_name)
+        host_ds_system = self.esxi.configManager.datastoreSystem
+        ds_system = vim.host.DatastoreSystem
+        error_message_mount = "Cannot mount datastore %s on host %s" % (self.datastore_name, self.esxi_hostname)
+        try:
+            vmfs_ds_options = ds_system.QueryVmfsDatastoreCreateOptions(host_ds_system,
+                                                                        ds_path,
+                                                                        self.vmfs_version)
+            vmfs_ds_options[0].spec.vmfs.volumeName = self.datastore_name
+            ds = ds_system.CreateVmfsDatastore(host_ds_system,
+                                               vmfs_ds_options[0].spec)
+        except vim.fault.NotFound:
+            self.module.fail_json(msg=error_message_mount + ": Device is not found")
+        except vim.fault.HostConfigFault:
+            self.module.fail_json(msg=error_message_mount + ": Unable to format the VMFS volume or gather information about the created volume")
+        except vim.fault.DuplicateName:
+            self.module.fail_json(msg=error_message_mount + ": A datastore with the same name already exists")
+        except vmodl.fault.InvalidArgument:
+            self.module.fail_json(msg=error_message_mount + ": The datastore name is invalid, or the spec is invalid")
+        except Exception as e:
+            self.module.fail_json(msg=error_message_mount + ": " + str(e))
+        self.module.exit_json(changed=True, result="datastore %s on host %s" % (self.datastore_name, self.esxi_hostname))
+
+
+def main():
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(
+        datacenter_name=dict(type='str', required=True),
+        datastore_name=dict(type='str', required=True),
+        datastore_type=dict(type='str', choices=['nfs', 'vmfs']),
```

Comment:
Add `required=True` (just a suggestion)

Reply:
Could be not required in case of umounting datastore.